src/coprocessor/dag/expr_util/like.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Implements SQL `LIKE`.
+//!
+//! This implementation needs refactor.
+//!
+//! 1. It is not effective. Consider target = 'aaaaaaaaaaaaaaa' and pattern = 'a%a%a%a%a%a%b'.
+//!    See https://research.swtch.com/glob
+//!
+//! 2. It should support non-binary mode (and binary mode) and do case insensitive comparing
+//!    in non-binary mode.
+
+use std::slice::Iter;
+
+use crate::coprocessor::dag::expr::Result;
+
+const MAX_RECURSE_LEVEL: usize = 1024;
+
+// Do match until '%' is found.
+#[inline]
+fn partial_like(tcs: &mut Iter<'_, u8>, pcs: &mut Iter<'_, u8>, escape: u32) -> Option<bool> {
+    loop {
+        match pcs.next().cloned() {
+            None => return Some(tcs.next().is_none()),
+            Some(b'%') => return None,
+            Some(c) => {
+                let (npc, escape) = if u32::from(c) == escape {
```

Comment:
This does not perform correctly with actual non-ASCII code points. Though it is rejected by TiDB anyway 🤷

MySQL 8:

```
mysql> select '1%4' like '1哈%4' escape '哈';
+----------------------------------+
| '1%4' like '1哈%4' escape '哈'   |
+----------------------------------+
|                                1 |
+----------------------------------+
1 row in set (0.01 sec)
```

TiDB 3.0:

```
mysql> select '1%4' like '1哈%4' escape '哈';
ERROR 1210 (HY000): Incorrect arguments to ESCAPE
```

Reply:
Thanks for pointing out. This is indeed yet another problem exists in current implementation. However this PR does not intend to fix them. Instead, it simply adapts it for the RPN framework so that TPC-H can work.